### PR TITLE
improve unnecessary_overrides

### LINKED ISSUE #626: unnecessary_override should special case noSuchMethod method
`noSuchMethod` has to be override in some cases:

```dart
abstract class A {
  m();
}
class B implements A {
  @override
  noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
}
```

Without the override I get a warning:

>Missing concrete implementation of 'A.m'

### PR DESCRIPTION
**Pull request by a14n**
Fixes #626, #627 and #747.

### CODE REVIEW

File: lib/src/rules/unnecessary_overrides.dart [hunk 1]
```diff
@@ -106,14 +120,27 @@ abstract class _AbstractUnnecessaryOverrideVisitor extends SimpleAstVisitor {
...
+
+  bool _haveSameDeclaration() {
+    if (declaration.element.returnType != inheritedMethod.returnType)
+      return false;
+    for (var i = 0; i < inheritedMethod.parameters.length; i++) {
+      final superParam = inheritedMethod.parameters[i];
+      final param = declaration.element.parameters[i];
+      if (param.type != superParam.type) return false;
+      if (param.name != superParam.name) return false;
+      if (param.isCovariant != superParam.isCovariant) return false;
```
Comment (bwilkerson) [inline lib/src/rules/unnecessary_overrides.dart]: We should check the optionality of the parameters. Should we also check the default value for option parameters?

File: lib/src/rules/unnecessary_overrides.dart [hunk 2]
```diff
@@ -54,12 +62,12 @@ class UnnecessaryOverrides extends LintRule {
 
 abstract class _AbstractUnnecessaryOverrideVisitor extends SimpleAstVisitor {
   LintRule rule;
-  Element inheritedMethod;
+  ExecutableElement inheritedMethod;
```
Comment (bwilkerson) [inline lib/src/rules/unnecessary_overrides.dart]: Consider making `inheritedMethod` and `declaration` be local variables in `visitMethodDeclaration` and parameters to `_haveSameDeclaration`.

Comment (a14n) [inline lib/src/rules/unnecessary_overrides.dart]: Those variables are used at several places and not only there.

File: lib/src/rules/unnecessary_overrides.dart (hunk 1 above)
Comment (a14n) [inline lib/src/rules/unnecessary_overrides.dart]: done

**Comment by a14n** [comment]
PTAL

**Pull request merged**